dotnet/src/Agents/Abstractions/AgentChat.cs

```diff
@@ -34,26 +50,51 @@ public abstract class AgentChat
     /// <param name="agent">An optional agent, if requesting an agent history.</param>
     /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
     /// <returns>The message history</returns>
-    public IAsyncEnumerable<ChatMessageContent> GetChatMessagesAsync(Agent? agent = null, CancellationToken cancellationToken = default)
+    public async IAsyncEnumerable<ChatMessageContent> GetChatMessagesAsync(
+        Agent? agent = null,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
     {
-        if (agent == null)
+        this.ThrowIfActive();
```

Comment:
This behaviour should be explained in the comments. I'm curious why we are doing this?

Reply:
Updated